lib/http1.c

```diff
@@ -994,15 +1008,27 @@ static h2o_iovec_t log_request_index(h2o_req_t *req)
     return h2o_iovec_init(s, len);
 }
 
+static int conn_is_h1(h2o_conn_t *conn)
+{
+    return conn->callbacks->get_sockname == get_sockname;
```

Comment:
How about promoting `static const h2o_conn_callbacks_t callbacks` defined in `h2o_http1_accept` to a file-level static variable and comparing its address with `conn->callbacks`?

Comparing the address of the callback table is more natural than comparing a particular value of a callback table.